trello/trelloclient.py

```diff
@@ -215,6 +216,9 @@ def fetch_json(
         if post_args is None:
             post_args = {}
 
+        # add user agent header to fetch_json requests
+        headers['User-Agent'] = generate_user_agent()
+
         # if files specified, we don't want any data
         data = None
         if files is None:
```

Comment:
I tested your code and it is still not work? Is it possible to add a condition to the post_args parameter, the relevant statement is as follows:

```python
        if files is None and post_args and post_args != {}:
            data = json.dumps(post_args)
```